spec/lib/supplier_import_spec.rb

```diff
@@ -0,0 +1,26 @@
+require 'rails_helper'
+require './lib/supplier_import'
+require './lib/supplier_import_row'
+
+RSpec.describe SupplierImport do
+  describe '#initialize' do
+    it 'accepts CSV data as a string input' do
+      csv = "frameworkreference,suppliername\nRM1234,Cheese & Crackers LLP\nRM5678,Vaughan Legal\n"
+
+      expect { SupplierImport.new(csv) }.to_not raise_error
+    end
+  end
+
+  describe '#run!' do
+    it 'calls SupplierImportRow.import! for each row contained in the CSV' do
+      # TODO: This should probably stub calls to SupplierImportRow
+
+      FactoryBot.create(:framework, short_name: 'RM1234')
+      FactoryBot.create(:framework, short_name: 'RM5678')
+
+      csv = "frameworkreference,suppliername\nRM1234,Cheese & Crackers LLP\nRM5678,Vaughan Legal\n"
+
+      expect { SupplierImport.new(csv).run! }.to change { Supplier.count }.by(2)
```

Comment:
just to make this a slightly more thorough spec, it might be worth verifying that the two new suppliers match the names and frameworks specified in the CSV. I know that means re-testing some of what's already tested in the specs for `SupplierImportRow`, but I think the cost of the overlap is minimal and the benefit in terms of giving us confidence that the pieces fit together makes it worth it.